assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java

```diff
@@ -996,12 +996,14 @@ public AssignmentSubmission addSubmission(String assignmentId, String submitter)
                 if (assignment.getIsGroup()) {
                     Group group = site.getGroup(submitter);
                     if (group != null && assignment.getGroups().contains(group.getReference())) {
-                        for (Member member : group.getMembers()) {
-                            AssignmentSubmissionSubmitter ass = new AssignmentSubmissionSubmitter();
-                            ass.setSubmitter(member.getUserId());
-                            submissionSubmitters.add(ass);
-                        }
-                        submission.setGroupId(submitter);
+                    	group.getMembers().stream()
+                    	.filter(m -> m.getRole().isAllowed(SECURE_ADD_ASSIGNMENT_SUBMISSION) || group.isAllowed(m.getUserId(), SECURE_ADD_ASSIGNMENT_SUBMISSION) )
```

Comment:
I think both checks seems to be the same (group isAllowed is calling role.isAllowed): 

https://github.com/sakaiproject/sakai/blob/master/kernel/kernel-impl/src/main/java/org/sakaiproject/authz/impl/BaseAuthzGroup.java#L762-L784

Do you think there are use cases that return different values @mpellicer ?

Reply:
Some permissions of Assignments are checked at a site level and at a group level, makes more sense to me to check at group level here.

Just saying, every time I worked with site and group permissions I faced this stuff, it's a bit messy.

Have you tested how it works after the changes @SedueRey ? Is it working for both cases? Is it working like the previous PR without the OR comparison?